### DISCUSSION TITLE
MQTT - Publisher

### DISCUSSION THREAD

**Discussion by bonello78** [netalertx/NetAlertX#646] [Q&A]
Hi everyone,
I have completed the integration between NetAlertX and my HA via MQTT and it works fine: a new device was detected on NAX and it created the device with its sensors on HA. Great!
Now I would like to understand how to expose all present devices not just the new ones on HA.

Thanks

**Comment by jokob-sk** [comment]
Selecting this setting will make sure devices are sent to MQTT & HA:

![image](https://github.com/jokob-sk/NetAlertX/assets/96159884/80914bdb-5275-4ff2-9851-779c19a58693)

You can check the event log to see exactly when and how devices are sent. In the log search for:

```
[Plugins] CMD: python3 /app/front/plugins/_publisher_mqtt/mqtt.py
```

The logs below the line will give you insights what gets synced. You can also check Integrations > PLugins > MQTT section in the UI.

Hope this helps,
j

**Reply by bonello78** [reply]
Thank you for your support. From the NAX log it seems that the information is sent correctly, from the MQTT log on HA I see that there is a connection (opened and then closed). Unfortunately, it does not create the devices and sensors. Any suggestions?

**Reply by jokob-sk** [reply]
You can try adjusting the `MQTT_VERSION` in the settings if that makes a difference. The next step would be checking if the devices are created on the broker, e.g. with [MQTT Explorer](https://mqtt-explorer.com/).

PLease also double-check the docs: https://github.com/jokob-sk/NetAlertX/blob/main/docs/HOME_ASSISTANT.md

**Reply by bonello78** [reply]
Thanks for your suggestions. I checked with MQTT Explorer and the NAX Sensors are in Home Assistant,
but they are grouped under the node **system-sensors** and I'm not able to see them in HA. Any suggestions?

![image](https://github.com/jokob-sk/NetAlertX/assets/95490780/505cfe8f-8a91-469c-b82b-0f742540eda3)

Thank you.

A.

**Reply by jokob-sk** [reply]
humm, maybe that's a question for HA forums. I use Mosquito and the sensors show up just fine.

![image](https://github.com/jokob-sk/NetAlertX/assets/96159884/a05bd296-59c4-407a-ad70-694f16053ffe)


![image](https://github.com/jokob-sk/NetAlertX/assets/96159884/68cce8fb-dbfe-424e-935e-f0040d7c0b21)


![image](https://github.com/jokob-sk/NetAlertX/assets/96159884/3043a7f4-18ad-45dc-839f-637572dbe675)

**Reply by jokob-sk** [reply]
Oh also, sensors show up for me in other sections:

![image](https://github.com/jokob-sk/NetAlertX/assets/96159884/f044afd4-25d5-4f0c-bedd-fa6d06e33f6d)
![image](https://github.com/jokob-sk/NetAlertX/assets/96159884/be029cae-8507-4d04-9b15-8ca81ae89bea)
![image](https://github.com/jokob-sk/NetAlertX/assets/96159884/566e34ab-9ae2-4a0e-9769-89f7158e83c7)

**Comment by ingoratsdorf** [comment]
Hi folks,

not sure what the issue is here, but the configs are under the base topic "homeassistant", which is correct. Those are really only the configs, not the states
Typically, there then is a user setting to provide another base topic for the actual sensor states, in our case "system-sensors".
I'd say it would be cool to have a user setting for the base topic, ie "NetAlertX sensors" or whatever. The config sections would then refer to that new base topic.
Example for user base topics are "openbeken", tasmota, etc. So "system" sensors may not be the best choice ;-)

But nonetheless, it should all work in HA. At least no issues ATM on my side.
